components/CurrencyCard.tsx

```diff
@@ -0,0 +1,69 @@
+import { StyleSheet, Text, View, Image } from 'react-native';
+import React, {FC, ReactElement, ReactNode} from 'react';
+
+import Icon from 'react-native-vector-icons/Feather'
+Icon.loadFont()
+
+const CurrencyCard = ({currency}: any) => {
+  
+
+  return (
+    <View style={styles.mainContainer}>
+      <View style={styles.leftContainer}>
+        <Image style={styles.icon} source={{uri: `https://messari.io/asset-images/${currency.id && currency.id}/128.png?v=2`}} />
```

Comment:
[Minor/Question] currency.id o currency puede ser null? si no es así ese and no es necesario, y en caso de que currency sea nulo o undefined, o la id la uri terminaría siendo:
 
`https://messari.io/asset-images//128.png?v=2`

hay que cambiar la forma en que se realiza ya sea retirando el and, utilizando un ternario o moviendo la / para dentro de la validación si es realmente necesaria

Reply:
Ese currency.id se refiere al objeto que vendrá después de la api, es que me adelanté y miré cómo venían los datos

Reply:
mmm si los datos pueden no llegarte conviene hacer una validacion fuera del componente antes de renderizarlo, y ya luego mantener limpio este, y que solo se valide una vez